app/src/main/java/ar/com/wolox/android/cookbook/mercadopago/MercadoPagoRecipePresenter.kt

```diff
@@ -0,0 +1,100 @@
+package ar.com.wolox.android.cookbook.mercadopago
+
+import ar.com.wolox.android.cookbook.BuildConfig
+import ar.com.wolox.android.cookbook.common.di.CoroutineDispatchersModule
+import ar.com.wolox.android.cookbook.common.utils.isValidEmail
+import ar.com.wolox.android.cookbook.mercadopago.model.Cart
+import ar.com.wolox.android.cookbook.mercadopago.model.Client
+import ar.com.wolox.android.cookbook.mercadopago.model.Product
+import ar.com.wolox.android.cookbook.mercadopago.network.MercadoPagoAdapter
+import ar.com.wolox.android.cookbook.mercadopago.network.MercadoPagoProductsRepository
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import com.mercadopago.android.px.core.MercadoPagoCheckout
+import com.mercadopago.android.px.model.Payment
+import com.mercadopago.android.px.model.exceptions.MercadoPagoError
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Named
+
+class MercadoPagoRecipePresenter @Inject constructor(
+    @Named(CoroutineDispatchersModule.MAIN) mainDispatcher: CoroutineDispatcher,
+    private val mercadoPagoProductsRepository: MercadoPagoProductsRepository,
+    private val mercadoPagoAdapter: MercadoPagoAdapter
+) : CoroutineBasePresenter<MercadoPagoView>(mainDispatcher) {
+
+    private var items: List<Pair<Product, Int>>? = null
+
+    override fun onViewAttached() {
+        view?.startLoading()
```

Comment:
Use `view?.run{}` instead

Reply:
Using that will add more kind of callback hell for none reason. IMO doing that would make the code less readable.